### DISCUSSION TITLE
Support for timezones

### DISCUSSION THREAD

**Discussion by Violet-Vibes** [wdelfuego/nova-calendar#24] [Ideas & Feature Requests]
Hi! I'm currently displaying the time of certain events using `$event->displayTime();` but as the users accessing the calendar all have different timezones, this can get confusing at times.

Inside my actual nova resource, I use this to display the time in the users timezone, 

`DateTime::make('Start At')->sortable()->required()->displayUsing(function ($value) use ($request) {
                return $value ? Carbon::parse($value)->setTimezone($request->user()->timezone)->toDateTimeString() : $value;
            }),`

Would it be possible (or it maybe already is and I just missed it) to show the times in the calendar in a similar fashion?

**Comment by wdelfuego** [comment]
You can overwrite both the start and end timestamps of each event in the `customizeEvent` method, so you could do timezone manipulation there.

I think adding this to `customizeEvent` should work in your case:

    $event->start($event->start()->setTimezone($this->request->user()->timezone));

You should do this for the `end()` timestamp, too, if you're using it.

You could get the user directly from the `Auth` facade or a global helper function but since you relied on the request and the CalendarDataProvider has a reference to it, I think this should be fine.

Having said that, perhaps some type of timezone support is required in the library. Right now, not even the `app.timezone` config variable is used. Maybe I'll add that as a default timezone and allow the implementation to override its value to do something per-user like you are doing.

**Comment by wdelfuego** [comment]
It was an easy fix, I have just released 1.5 which makes this a little easier for you. If you `composer update` all you should have to do is implement `timezone()` in your CalenderDataProvider, so you won't have to fiddle with the start and end dates yourself.

See the [documentation](https://wdelfuego.github.io/nova-calendar/customizing-the-calendar.html), too.

**Reply by Violet-Vibes** [reply]
> It was an easy fix, I have just released 1.5 which makes this a little easier for you. If you `composer update` all you should have to do is implement `timezone()` in your CalenderDataProvider, so you won't have to fiddle with the start and end dates yourself.
> 
> See the [documentation](https://wdelfuego.github.io/nova-calendar/customizing-the-calendar.html), too.

This works wonders, thank you for implementing this so quick! :)
